### ISSUE TITLE
Releasing Neural Network on Microcontroller (NNoM) using CMSIS-NN as backend

### ISSUE THREAD

**Issue by majianjia** [ARM-software/CMSIS_5#530]
Hi All,

For those who are struggling with the massive parameters in CMSIS-NN,

I just released NN framework called Neural Network on Microcontroller (NNoM).
Currently, NNoM runs on top of the highly optimized CMSIS-NN. It provides a higher-level interface similar to Keras. Allows you to try state-of-the-art structures such as Inception ([example](https://github.com/majianjia/nnom/tree/master/examples/uci-inception)), ResNet, DenseNet([example](https://github.com/majianjia/nnom/tree/master/examples/mnist-densenet)) ... with very little effort. 

The idea is, instead of writing plenty of scripts to generate C codes from a complex NN model, we write it directly in C codes with similar APIs. 
NNoM is a layer-based framework as same as Keras, allows building complex structures like Inception and ResNet or simple sequential model in minutes. Please check the examples [ Inception example using UCI HAR dataset](https://github.com/majianjia/nnom/blob/b8798c6e4e65a17addfa597cbb042c0ad0edc3f3/examples/uci-inception/mcu/main.c#L120)

The model summary will be printed while compiling. It also provides some evaluation metrics such as Top-K accuracy and Confusion Matrix, allows you to validate the implementation on MCU directly.  Runtime stat is also provided when a us timer is ported. 

Currently, it supports 8-bit quantized only. 
Supports Sequential and Functional models. 
Supports 1/2D in Conv, Depthwise Conv, Dense, Softmax
Supports Lambda layer. 
Supports Average Pooling, Max Pooling, Global Avg/Max Pooling
Supports ReLU, TanH, Sigmoid
Supports Matrix  Multiplier, Addition, Subtraction, Concatenate in any axis.

For more detail please check the repo:
https://github.com/majianjia/nnom

Thanks for your attention,
Jianjia Ma
